internal/work/results/shards.go

```diff
@@ -0,0 +1,109 @@
+package results
+
+import (
+	"fmt"
+
+	"github.com/eigerco/strawberry/internal/common"
+	"github.com/eigerco/strawberry/internal/crypto"
+	"github.com/eigerco/strawberry/internal/erasurecoding"
+	"github.com/eigerco/strawberry/internal/merkle/binary_tree"
+	"github.com/eigerco/strawberry/internal/work"
+	"github.com/eigerco/strawberry/pkg/serialization/codec/jam"
+)
+
+type Shards struct {
+	Bundle                    [][]byte
+	Segments                  [][][]byte // in case no segments are present this could be nil
```

Comment:
Starting to wonder if it wouldn't be better with a `Shard` type that's [][]byte. But yeah.

Reply:
Yeah, I also thought so initially, but it having it like this has an advantage, we can pass parameters of type `[][]byte`  directly to functions that require it, instead of having to convert from `[]Shard.*.Bundle`